scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -641,52 +658,65 @@ function Remove-SecurityDaemonResources {
     else {
         Write-Verbose "$cmdErr"
     }
+    
+	# Check whether we need to clean up after an errant installation into the OS partition on IoT Core
+	if ($env:ProgramData -ne 'C:\ProgramData') {
+		Write-Verbose "Multiple ProgramData directories found"
+		$existingMobyDataRoots = $MobyDataRootDirectory, $MobyStaticDataRootDirectory | ? {test-path $_}
```

Comment:
`| ?{ Test-Path $_ }` (Same comment as before.)

Reply:
Alright, I've removed it as it's now redundant and this check will prevent us from throwing garbage into the verbose install log on desktop installations.

Reply:
You need a `Test-Path` *somewhere*, either here or in the `foreach ($root in $existingMobyDataRoots) {` loop. As the code is right now, it'll try to delete the directories even if they don't exist and end up setting `$success` to false.